pkg/rtc/mediatrack.go

```diff
@@ -136,7 +136,7 @@ func (t *MediaTrack) SetMuted(muted bool) {
 }
 
 func (t *MediaTrack) OnClose(f func()) {
-	t.onClose = f
+	t.onClose = append(t.onClose, f)
```

Comment:
Does this need a lock?

Reply:
it only called by created in participant, then called by RelayManager AddUpTrack, so lock is no need.

Reply:
In a couple of places, we would set OnClose to nil. those wouldn't be handled correctly. I suggest renaming this to `AddOnClose()` and removing the cases we are setting it to nil

Reply:
Done